### PR TITLE
test: add e2e tests for gateway-api form edge cases

### LINKED ISSUE #715: Add e2e tests for gateway-api form edge cases
## Description

PR #712 fixed pre-existing bugs in gateway-api form components (issue #692). The existing e2e tests only cover happy paths and need new tests to cover the edge cases that were fixed.

## Changes to cover

- All 9 HTTP methods available in method dropdown (HEAD, CONNECT, OPTIONS, TRACE were added)
- Request header filter creation (currently skipped entirely in e2e tests)
- Port validation with `Number.isInteger` and 1-65535 bounds check
- Parent ref serialization without sectionName

## Proposed tests

Add to `e2e/tests/httproute-crud.spec.ts` as `@nightly`:

1. **All HTTP methods in dropdown** - verify all 9 methods appear, create route using OPTIONS, verify via kubectl
2. **Create HTTPRoute with request header filter** - add RequestHeaderModifier filter with header name and empty value, verify filter appears in created resource

## Related

- Fixes from #692 (PR #712)

### PR DESCRIPTION
**Pull request by emmaaroche**
## Description

Adds e2e test coverage for gateway-api form edge cases fixed in #712 (issue #692).

Closes #715

## Changes

- Add test for all 9 HTTP methods in dropdown (GET, HEAD, POST, PUT, DELETE, CONNECT, OPTIONS, TRACE, PATCH)
- Add test for request header filter creation
- Add test for port validation in mirror filter
- Add test for parentRef serialization without sectionName
- Update `addRuleViaWizard` helper to accept optional `method` parameter

## Testing

All 4 new tests pass locally:

```
✓ all 9 HTTP methods available in method dropdown @nightly
✓ creates HTTPRoute with request header filter @nightly
✓ preserves parentRef without sectionName on edit @nightly
✓ port validation accepts valid port in mirror filter @nightly
```

Each test validates actual resource state via kubectl after form submission.

## Summary by CodeRabbit

- **New Features**
  - HTTP route rules now support all nine standard HTTP methods.
  - Request header filters can be configured for more precise routing.
  - Request mirroring supports valid port configurations.

- **Bug Fixes**
  - Editing HTTP route rules preserves the parent route reference, even when no section name is provided.

### CODE REVIEW

File: e2e/tests/httproute-crud.spec.ts [hunk 1]
```diff
@@ -311,4 +317,372 @@ spec:
...
+    // Next → Filters
+    await page.getByRole('button', { name: 'Next' }).click();
+
+    // Add a RequestHeaderModifier filter
+    await page.getByRole('button', { name: 'Add filter' }).click();
+
+    // Fill header name and value
+    await page.locator('#hdr-name-0').fill('X-Custom-Header');
+    await page.locator('#hdr-value-0').fill('test-value');
+
```
[the remark this answers is not in the document]
Comment (emmaaroche) [reply e2e/tests/httproute-crud.spec.ts:438]: Can't test empty header values - the HTTPRoute CRD spec requires header values to be at least 1 character long. When I tried clearing the value, the Kubernetes API server rejected the resource with:

`Error "Invalid value: "": spec.rules[0].filters[0].requestHeaderModifier.add[0].value in body should be at least 1 chars long"`

The form would need to validate and block empty values before submission to prevent this API error, but testing that the persisted resource has an empty value isn't possible since the API won't accept it.

**Comment by R-Lawton** [comment]
👀

**Review by R-Lawton — COMMENTED** [review]
nice set of edge-case tests. couple of things i think are worth addressing:

**unused helper param** (`e2e/tests/httproute-crud.spec.ts:78-87`) — the `method` param added to `addRuleViaWizard` isn't called by any of the new tests, they all inline the wizard steps. can we either wire it into one of them (the all-9-methods test seems like a natural fit) or drop it so we're not carrying unused code?

**duplication** (`httproute-crud.spec.ts:321+, 413+, 497+, 596+`) — the ~13-line create + parentRef setup block is repeated across all four tests. worth pulling into a `setupCreateRouteWithParent` helper so it's easier to maintain?

**Review by Anton-Fil — APPROVED** [review]
LGTM
